opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java

```diff
@@ -163,6 +168,18 @@ public void findLastEvent(StructureTaskDetails taskDetails) {
 
     }
 
+    @Override
+    public void resetTaskInfo(Context context, StructureTaskDetails taskDetails) {
+        appExecutors.diskIO().execute(() -> {
+            TaskUtils.getInstance().resetTask(context, taskDetails);
+            interactorUtils.archiveEventsForTask(getDatabase(), taskDetails);
+
+            appExecutors.mainThread().execute(() -> {
+                presenter.onOnTaskInfoReset(taskDetails.getStructureId());
+            });
+        });
+    }
+
```

Comment:
Move to InteractorUtils as its similar to method in TaskRegisterFragmentInteractor